OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -28,9 +28,18 @@ public override Activity Tick(Actor self)
 			if (!target.IsValidFor(self))
 				return NextActivity;
 
-			var limitedAmmo = self.TraitOrDefault<LimitedAmmo>();
-			if (limitedAmmo != null && !limitedAmmo.HasAmmo())
-				return NextActivity;
+			var limitedAmmo = self.TraitsImplementing<LimitedAmmo>();
+			if (limitedAmmo != null)
+			{
+				foreach (var la in self.TraitsImplementing<LimitedAmmo>())
+				{
+					// Skip LimitedAmmo pools that SelfReload or still have ammo)
+					if (la.ReloadsAutomatically() || la.HasAmmo())
+						continue;
+
+					return NextActivity;
+				}
+			}
```

Comment:
Oh, I see.
If I remember correctly (which isn't guaranteed, given my poor memory), we use brackets when there's more than 1 line following the ```if```, for the sake of readability.